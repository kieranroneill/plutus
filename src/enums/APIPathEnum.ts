enum APIPathEnum {
  Chains = 'chains',
  Fees = 'fees',
  Versions = 'versions',
}

export default APIPathEnum;
