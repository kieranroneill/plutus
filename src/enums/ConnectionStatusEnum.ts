enum ConnectionStatusEnum {
  Connected = 'connected',
  Connecting = 'connecting',
  Disconnected = 'disconnected',
  Disconnecting = 'disconnecting',
  Uninitialized = 'uninitialized',
}

export default ConnectionStatusEnum;
