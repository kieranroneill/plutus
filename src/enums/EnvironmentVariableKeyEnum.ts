enum EnvironmentVariableKeyEnum {
  // base
  AppName = 'APP_NAME',
  AppPort = 'APP_PORT',
  AppVersion = 'APP_VERSION',
  LogLevel = 'LOG_LEVEL',
  NodeEnv = 'NODE_ENV',

  // database
  MongoDBHost = 'MONGODB_HOST',
  MongoDBName = 'MONGODB_NAME',
  MongoDBPassword = 'MONGODB_PASSWORD',
  MongoDBPort = 'MONGODB_PORT',
  MongoDBUsername = 'MONGODB_USERNAME',
}

export default EnvironmentVariableKeyEnum;
