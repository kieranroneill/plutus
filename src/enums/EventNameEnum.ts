enum EventNameEnum {
  FeesCollectedEventQuery = 'fees_collected_event.query',
  FeesCollectedEventNew = 'fees_collected_event.new',
}

export default EventNameEnum;
