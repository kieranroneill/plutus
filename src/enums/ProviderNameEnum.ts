enum ProviderNameEnum {
  DatabaseConnection = 'DatabaseConnection',
  FeeModel = 'FeeModel',
}

export default ProviderNameEnum;
