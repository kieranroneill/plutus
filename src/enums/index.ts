export { default as APIPathEnum } from './APIPathEnum';
export { default as EnvironmentVariableKeyEnum } from './EnvironmentVariableKeyEnum';
