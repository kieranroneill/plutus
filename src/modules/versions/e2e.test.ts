import { HttpStatus } from '@nestjs/common';
import { agent as request, Response, Agent } from 'supertest';

// enums
import { APIPathEnum } from '@app/enums';

describe(`/${APIPathEnum.Versions}`, () => {
  let agent: Agent;

  beforeAll(async () => {
    agent = request(`http://127.0.0.1:3000`);
  });

  describe(`GET /${APIPathEnum.Versions}`, () => {
    it('should get the api info', async () => {
      const response: Response = await agent
        .get(`/${APIPathEnum.Versions}`)
        .expect(HttpStatus.OK);

      expect(response.body).toMatchSnapshot();
    });
  });
});
