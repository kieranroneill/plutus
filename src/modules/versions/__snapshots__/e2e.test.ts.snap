// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/versions GET /versions should get the api info 1`] = `
{
  "databases": [
    {
      "status": "connected",
      "type": "mongodb",
      "version": "8.2.1",
    },
  ],
  "environment": "test",
  "name": "plutus-test",
  "version": "1.0.0",
}
`;
